### PR TITLE
Time external cache response time

### PR DESCRIPTION
**Pull request by erikjohnston**
It looks like on matrix.org this can take a surprisingly long time.

### CODE REVIEW

File: synapse/replication/tcp/external_cache.py [hunk 1]
```diff
@@ -72,23 +86,25 @@ async def set(self, cache_name: str, key: str, value: Any, expiry_ms: int) -> No
 
         logger.debug("Caching %s %s: %r", cache_name, key, encoded_value)
 
-        return await make_deferred_yieldable(
-            self._redis_connection.set(
-                self._get_redis_key(cache_name, key),
-                encoded_value,
-                pexpire=expiry_ms,
+        with response_timer.labels("set").time():
```
Comment (clokep) [inline synapse/replication/tcp/external_cache.py:89]: Would it make sense to add the `cache_name` as a label?

Comment (erikjohnston) [reply synapse/replication/tcp/external_cache.py:89]: I don't think so, the times shouldn't depend on what we're caching (and I'm a bit cautious about the fact it adds a bunch of extra metrics). Broadly these should be super quick, and if they're not then something is wrong/congested.

File: synapse/replication/tcp/external_cache.py [hunk 2]
```diff
@@ -35,6 +35,20 @@
     labelnames=["cache_name", "hit"],
 )
 
+response_timer = Histogram(
+    "synapse_external_cache_response_time",
+    "sec",
```
Comment (richvdh) [inline synapse/replication/tcp/external_cache.py:40]: better documentation please

File: synapse/replication/tcp/external_cache.py [hunk 3]
```diff
@@ -35,6 +35,20 @@
     labelnames=["cache_name", "hit"],
 )
 
+response_timer = Histogram(
+    "synapse_external_cache_response_time",
```
Comment (richvdh) [inline synapse/replication/tcp/external_cache.py:39]:
```suggestion
    "synapse_external_cache_response_time_seconds",
```

**Review by richvdh — CHANGES_REQUESTED** [review]

File: synapse/replication/tcp/external_cache.py [hunk 4]
```diff
@@ -36,8 +36,8 @@
 )
 
 response_timer = Histogram(
-    "synapse_external_cache_response_time",
-    "sec",
+    "synapse_external_cache_response_time_seconds",
+    "Time taken to get a response from Redis for a request",
```
Comment (richvdh) [inline synapse/replication/tcp/external_cache.py:40]:
```suggestion
    "Time taken to get a response from Redis for a cache get/set request",
```

or something

**Review by richvdh — APPROVED** [review]
lgtm

**Pull request merged**